src/components/UserSelector.tsx

```diff
@@ -1,45 +1,41 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { clearPost } from '../features/Slices/selectedPostSlice';
+import { setAuthor } from '../features/Slices/author';
+import * as usersActions from '../features/Slices/usersSlice';
+export const UserSelector: React.FC = () => {
+  const [expanded, setExpanded] = useState(false);
+  const users = useAppSelector(state => state.users.users);
+  const author = useAppSelector(state => state.author.author);
+  const dispatch = useAppDispatch();
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
+  const selectAuthor = (user: User) => {
+    dispatch(setAuthor(user));
+    dispatch(clearPost());
+  };
 
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
-  const [expanded, setExpanded] = useState(false);
+  useEffect(() => {
+    if (users.length === 0) {
+      dispatch(usersActions.loadUsers());
+    }
+  }, [dispatch, users.length]);
 
   useEffect(() => {
     if (!expanded) {
       return;
     }
 
-    // we save a link to remove the listener later
     const handleDocumentClick = () => {
-      // we close the Dropdown on any click (inside or outside)
-      // So there is not need to check if we clicked inside the list
       setExpanded(false);
     };
 
     document.addEventListener('click', handleDocumentClick);
 
-    // eslint-disable-next-line consistent-return
     return () => {
       document.removeEventListener('click', handleDocumentClick);
     };
-    // we don't want to listening for outside clicks
-    // when the Dopdown is closed
   }, [expanded]);
```

Comment:
This useEffect is used for handling outside click for the dropdown. It's a good practice to handle outside click in a dropdown component, but this implementation could be improved. Instead of adding and removing event listener on each render, you can add it once when the component mounts and remove it when the component unmounts. You can use the expanded state inside the event handler directly. If you want the event handler to always have the current state, you can use useRef to store the state.